packages/core/src/components/tabs/_overrides.scss

```diff
@@ -2,30 +2,225 @@
+
+We need to win the specificity of the vertical selector that sets box-shadow: none (declared in the _tabs.scss file):
+.#{$ns}-tabs.#{$ns}-vertical {
+(...)
+  > .#{$ns}-tab-list {
+    (...)
+    .#{$ns}-tab {
+      (...)
+      &[aria-selected="true"] {
+        box-shadow: none; => we need to override this
+      }
+    }
+    (...)
+}
+
+That's why we use (for example) this selector for vertical tabs: .#{$ns}-g-stroke-line-top.#{$ns}-tabs.#{$ns}-vertical.
+*/
+
+$stroke-line-top: inset 0 ($tab-indicator-width);
+$stroke-line-right: inset (-$tab-indicator-width) 0;
+$stroke-line-bottom: inset 0 (-$tab-indicator-width) 0;
+$stroke-line-left: inset $tab-indicator-width 0;
+
+@mixin strokeLine ($strokeLinePosition) {
   .#{$ns}-tab {
-    flex: 1;
-    margin: 0;
-    text-align: center;
-    border-bottom: 2px solid transparent;
+    &[aria-selected="true"] {
+      box-shadow: $strokeLinePosition !important;
+    }
+    &:not([aria-disabled="true"]):not(.#{$ns}-g-tab-disable-hover-effect):hover {
+      box-shadow: $strokeLinePosition $pt-intent-primary !important;
+    }
+  }
+}
 
-    &:not([aria-selected="true"]) {
-      border-bottom-color: $dark-gray1;
+/*  Here we are setting the default values if the position of the stroke line is not defined.
+    - vertical list => stroke line left
+    - horizontal list => stroke line bottom
+*/
+
+// light theme & dark theme
+.#{$ns}-dark,
+.#{$ns}-tab-list {
+  @include strokeLine($stroke-line-bottom);
+}
+
+//vertical tabs
+.#{$ns}-tabs.#{$ns}-vertical {
+  > .#{$ns}-tab-list {
+    @include strokeLine($stroke-line-left);
+  }
+}
+
+/*
+  Vertical Tabs when animated:
+  In BP, when vertical is used, the active tab is shown with background-color
+  But here we need that the stroke animated line is shown in the desired theme color.
+*/
+
+.#{$ns}-tabs.#{$ns}-vertical {
+  > .#{$ns}-tab-list {
+    .#{$ns}-tab-indicator-wrapper .#{$ns}-tab-indicator {
+      /* make stroke line vertical left */
+      height: inherit;
+      width: $tab-indicator-width;
+      /*overwrite BP styles*/
+      border-radius: 0;
+      background-color: $tab-color-selected;
     }
   }
 }
+
+
+.#{$ns}-dark {
+  .#{$ns}-tabs.#{$ns}-vertical {
+    > .#{$ns}-tab-list {
+      .#{$ns}-tab-indicator-wrapper .#{$ns}-tab-indicator {
+        background-color: $dark-tab-color-selected;
+      }
+    }
+  }
+}
+
+/*
+  Stroke line position declaration.
+  Available classNames to place the stroke line in the desired position:
+    - top => .#{$ns}-g-stroke-line-top => example: bp3-g-stroke-line-top
+    - right => .#{$ns}-g-stroke-line-right => example: bp3-g-stroke-line-right
+    - bottom => .#{$ns}-g-stroke-line-bottom => example: bp3-g-stroke-line-bottom
+    - left => .#{$ns}-g-stroke-line-left => example: bp3-g-stroke-line-left
+*/
+
+@mixin disableAnimation() {
+  .#{$ns}-tab-indicator-wrapper .#{$ns}-tab-indicator {
+    background-color: transparent;
+  }
+}
```

Comment:
Why something called `disableAnimation` hide a background-color ? Is it possible to want the `tab-indicator` visible but not animation?

I don't want to make the code more complex than necessary ... just wondering if the names need to be updated or I'm missing something.

Reply:
Good point!!  at least the mixin name expressed what I wanted 😆. 
Also, I was using the camel case style for the mixin name but I notice that BP uses the snake case, so I'm going to change it to:
```css
/*
  We need to hide the animated stroke line when we are handling one of the 'unsupported' stroke line positions. 
  The supported cases are:
  - horizontal => bottom line
  - vertical => left line
  In the other positions, we will need to hide the animated bar.
*/
@mixin hide-animated-stroke-line() {
  .#{$ns}-tab-indicator-wrapper {
    display: none;
  }
}

```